source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -2516,431 +2642,414 @@ GLSSharpNavierStokesSolver<dim>::sharp_edge()
+                      bool use_ib_for_pressure =
+                        (dof_is_inside) && (component_i == dim) &&
+                        (this->simulation_parameters.particlesParameters
+                           ->assemble_navier_stokes_inside == false) &&
+                        count_particles < 2;
+
+                      // Check if the DOfs is owned and if it's not a hanging
+                      // node.
+                      if (((component_i < dim) || use_ib_for_pressure) &&
+                          this->locally_owned_dofs.is_element(
+                            global_index_overwrite) &&
+                          (this->zero_constraints.is_constrained(
+                             local_dof_indices[i]) ||
+                           this->nonzero_constraints.is_constrained(
+                             local_dof_indices[i])) == false)
                         {
-                          // If we are here, the DOF is on a boundary.
-                          particle_close_to_wall = true;
-                          cell_2                 = cell;
-                          // If a boundary condition is already applied to this
-                          // DOF we skip it otherwise we impose a value base on
-                          // the velocity of the particle.
-                          if (this->zero_constraints.is_constrained(
-                                global_index_overwrite) ||
-                              this->nonzero_constraints.is_constrained(
-                                global_index_overwrite))
+                          // We are working on the velocity of the cell cut
+                          // loops on the dof that are for vx or vy separately
+                          // loops on all the dof of the cell that represent
+                          // a specific component
+                          // Define which dof is going to be redefined
+
+                          // Clear the current line of this dof
+                          this->system_matrix.clear_row(global_index_overwrite);
+
+                          // Define the points for the IB stencil, based on the
+                          // order and the particle position as well as the DOF
+                          // position. Depending on the order, the output
+                          // variable "point" change definition. In the case of
+                          // stencil orders 1 to 4 the variable point returns
+                          // the position of the DOF directly. In the case of
+                          // high order stencil, it returns the position of the
+                          // point that is on the IB. The variable
+                          // "interpolation points" return the points used to
+                          // define the cell used for the stencil definition and
+                          // the locations of the points use in the stencil
+                          // calculation.
```

Comment:
This is weird right? Why does the function change output as a function of inputs in such a dramatic fashion?